.github/workflows/release.yml

```diff
@@ -12,7 +12,7 @@ jobs:
       contents: read
     runs-on: ubuntu-latest
     env:
-      NX_CLOUD_ACCESS_TOKEN: ${{ secrets.NX_CLOUD_ACCESS_TOKEN }}
+      NX_CLOUD_ACCESS_TOKEN: ${{ !(github.event_name == 'push' && github.ref != 'main') && secrets.NX_CLOUD_ACCESS_TOKEN || '' }}
```

Comment:
it will always be `''` because `github.event_name` can only be `release` if think 
as the workflow is run `on` that specific event (see line 4)